Didstopia.PDFSharp.Tests/PDFSharpTests.cs

```diff
@@ -64,31 +66,51 @@ public void TestBasicFunctionality()
             loadedPdfDocument.Dispose();
         }
 
-        [Fact(Skip = "Skipping encryption tests due to a pending issue: https://github.com/Didstopia/PDFSharp/issues/3")]
-        public void TestEncryption()
+        /// <summary>
+        /// Ensure opening documents with null provider, or invalid password fail predicably
+        /// </summary>
+        [Fact()]
+        public void TestEncryptionFailsProperly()
         {
-            // FIXME: This is still breaking due to an NRE being thrown in the security classes/methods
-
-            // Opening a document will fail with an invalid password.
-            try
-            {
                 //var document = PdfReader.Open(PasswordSamplePath, "invalid password");
-                var document = PdfReader.Open(PasswordSamplePath, PdfDocumentOpenMode.ReadOnly, null);
-            }
-            catch (Exception ex)
-            {
-                Console.WriteLine(ex);
+                var document = PdfReader.Open(PasswordSamplePath, PdfDocumentOpenMode.ReadOnly, null);
+               
+                Assert.Throws<PdfReaderException>(() =>
+                {
+                    document = PdfReader.Open(PasswordSamplePath, "invalid password", PdfDocumentOpenMode.ReadOnly);
+                });
+
+
+            try
+            {
+                document = PdfReader.Open(PasswordSamplePath, "invalid password", PdfDocumentOpenMode.ReadOnly);
             }
+            
+            catch (Exception ex)
+            {
+                Assert.Equal(ex.Message, "The specified password is invalid.");

```

Comment:
These two parameters should be flipped, so it matches the assertion logic, but also to get rid of the following warning:

```
PDFSharpTests.cs(91,17): warning xUnit2000: The literal or constant value "The specified password is invalid." should be the first argument in the call to 'Assert.Equal(expected, actual)' in method 'TestEncryptionFailsProperly' on type 'PDFSharpTests'. [/home/travis/build/Didstopia/PDFSharp/Didstopia.PDFSharp.Tests/Didstopia.PDFSharp.Tests.csproj]
```